api/soap/mc_issue_api.php

```diff
@@ -436,8 +436,7 @@ function mci_issue_set_monitors( $p_issue_id, $p_requesting_user_id, array $p_mo
 	# 2. build new monitors ids
 	$t_new_monitor_ids = array();
 	foreach ( $p_monitors as $t_monitor ) {
-		$t_monitor = SoapObjectsFactory::unwrapObject( $t_monitor );
-		$t_new_monitor_ids[] = $t_monitor['id'];
+		$t_new_monitor_ids[] = mci_get_user_id( $t_monitor );
```

Comment:
@rombert 
As a side note. I think mci_get_user_id should be changed.
This piece of code has been introduced since fix of https://www.mantisbt.org/bugs/view.php?id=13571
```php
	} elseif( isset( $p_user['email'] ) ) {
		$t_user_id = user_get_id_by_email( $p_user['email'] );
	}
```
I think it was not a good decision to merge it as email is not unique.